tests/docs/test_docs.py

```diff
@@ -9,12 +11,49 @@
 
 def test_readme():
     """The blocks in the README become a valid python script when concatenated."""
-    readme_code = extract_code_blocks("README.md")
+    readme_code = "\n".join(extract_code_blocks("README.md"))
     exec(readme_code)
 
 
 @pytest.mark.parametrize("file", Path("docs/userguide/").rglob("*.md"))
 def test_userguide(file: Path):
     """The blocks in the user guide become a valid python script when concatenated."""
-    userguide_code = extract_code_blocks(file)
+    userguide_code = "\n".join(extract_code_blocks(file))
     exec(userguide_code)
+
+
+@pytest.mark.parametrize("file", Path("docs/userguide/").rglob("*.md"))
+def test_userguide_code_format(file: Path):
+    """The blocks in the user guide are properly formatted.
+
+    If it fails, run `pytest` with the `-s` flag to show the necessary format changes.
```

Comment:
What does pytest with -s flag do in general? Also, we should put this information in the README file of the tests as this is deeply hidden if we do not put it there.

Reply:
`-s` just unblocks the `print` messages from your code (which are not shown in a regular pytest run). Do you really think this is necessary, i.e. will external contributors ever write doc pages? I mean, if you have a good idea how to mention it, please do so :D I don't really have an idea, because this is such a special context. In case people really write a userguide and do not format their code properly, this test will fail and they will look at the test definition, I think!?

Reply:
If that is just a default feature of pytest, then I do not think that it is necessary to put this into the README with a link to the test. We might however still want to include a general sentence in the README like "Note that pytest suppresses all print messages, which can be unblocked using the `-s` option" if you feel that this information is valuable. If not, feel free to resolve :)

Reply:
Would actually skip this because this is general pytest stuff. That would be like us describing "note that numpy arrays can be indexed with brackets" ... not our responsibility 😃